extensions/smallrye-graphql/deployment/src/test/java/io/quarkus/smallrye/graphql/deployment/UniTest.java

```diff
@@ -0,0 +1,153 @@
+package io.quarkus.smallrye.graphql.deployment;
+
+import static io.quarkus.smallrye.graphql.deployment.AbstractGraphQLTest.MEDIATYPE_JSON;
+import static io.quarkus.smallrye.graphql.deployment.AbstractGraphQLTest.getPropertyAsString;
+
+import java.time.LocalDate;
+import java.time.Month;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.microprofile.graphql.GraphQLApi;
+import org.eclipse.microprofile.graphql.Query;
+import org.eclipse.microprofile.graphql.Source;
+import org.hamcrest.CoreMatchers;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+import io.restassured.RestAssured;
+import io.restassured.filter.log.ResponseLoggingFilter;
+import io.smallrye.mutiny.Uni;
+
+/**
+ * Make sure that Uni works
+ */
+public class UniTest extends AbstractGraphQLTest {
+
+    @RegisterExtension
+    static QuarkusUnitTest test = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClasses(BookGraphQLApi.class, Book.class)
+                    .addAsResource(new StringAsset(getPropertyAsString()), "application.properties")
+                    .addAsManifestResource(EmptyAsset.INSTANCE, "beans.xml"));
+
+    @Test
+    public void testSourcePost() {
+        String booksRequest = getPayload("{\n" +
+                "    book(name:\"Lord of the Flies\"){\n" +
+                "       title\n" +
+                "       published\n" +
+                "       buyLink\n" +
+                "       authors {\n" +
+                "           name\n" +
+                "           bornName\n" +
+                "       }\n" +
+                "    }\n" +
+                "}");
+
+        RestAssured.given()
+                .filter(new ResponseLoggingFilter())
+                .when()
+                .accept(MEDIATYPE_JSON)
+                .contentType(MEDIATYPE_JSON)
+                .body(booksRequest)
+                .post("/graphql")
+                .then()
+                .assertThat()
+                .statusCode(200)
+                .and()
+                .body(CoreMatchers.containsString("Lord of the Flies"),
+                        CoreMatchers.containsString("https://www.amazon.com/s?k=Lord+of+the+Flies&i=stripbooks-intl-ship"), // Test source
+                        CoreMatchers.containsString("William Gerald Golding")); // Test batch
+
+    }
+
+    @GraphQLApi
+    public static class BookGraphQLApi {
+
+        @Query
+        public Uni<Book> getBook(String name) {
+            return Uni.createFrom().item(BOOKS.get(name));
```

Comment:
These methods are returning Uni but are not async, they compute the emitted item synchronously. While it's ok in tests, it's NOT how things should work in any example and documentation.

Reply:
Can you help with the correct example ?

Reply:
Ok so () -> BOOKS.get(name) ?

Reply:
Done